test/integration.js

```diff
@@ -0,0 +1,204 @@
+			new assert.Assertion(text).equals('Timeout or not connected to KeePassXC', 'Error message is correct');
+		});
+	});
+
+	// Connecting to native messaging host org.keepassxc.keepassxc_browser (keepass.js:870)
+	test.it('Test reload button', function() {
+		wd.get(POPUP_URL);
+
+		wd.findElement({id: 'reload-status-button'}).then(function(reloadButton) {
+			new assert.Assertion(reloadButton.isDisplayed(), 'Reload button is displayed');
+
+			wd.actions().click(reloadButton).perform().then(function() {
+				// Check here that the functionality actually does something
+
+			});
+		});
+	});
+});
+
+// This test is disabled because AFAIK there's no way to get localStorage of an extension
+// Test options page and check the settings are enabled/disabled in localStorage
+/*test.describe('Testing options page', function() {
+	test.it('Read default settings', function() {
+		wd.get(OPTIONS_URL);
+
+		var checkOptions = [
+			'checkUpdateKeePassXC',
+			'autoCompleteUsernames',
+			'autoFillAndSend',
+			'usePasswordGenerator',
+			'autoFillSingleEntry',
+			'autoRetrieveCredentials'
+			'showNotifications'
+		];
+
+		var numberOptions = [
+			'blinkTimeout',
+			'blinkMinTimeout',
+			'allowedRedirect'
+		];
+
+		checkOptions.forEach(function(option) {
+			var defaultOption = localStorage[option];
+
+			wd.actions().click(wd.findElement({name: option})).perform().then(function() {
+				console.log(localStorage[option] + ' - ' + defaultOption);
+				new assert.Assertion(localStorage[option] !== defaultOption, "Option test: " + option);
+				wd.actions().click(wd.findElement({name: option})).perform().then(function() {
+					console.log(localStorage[option] + ' - ' + defaultOption);
+					new assert.Assertion(localStorage[option] === defaultOption, "Option test: " + option);
+				});
+			});
+		});
+
+		numberOptions.forEach(function(option) {
+			var defaultOption = localStorage[option];
+			console.log(defaultOption);
+			var buttonName = option + 'Button';
+
+			var numberField = wd.findElement({id: option});
+			numberField.sendKeys('2').then(function() {
+				wd.actions().click(wd.findElement({id: buttonName})).perform().then(function() {
+					console.log(localStorage[option] + ' - ' + defaultOption);
+					new assert.Assertion(localStorage[option] !== defaultOption, "Option test: " + option);
+					wd.actions().click(wd.findElement({id: buttonName})).perform().then(function() {
+						console.log(localStorage[option] + ' - ' + defaultOption);
+						new assert.Assertion(localStorage[option] === defaultOption, "Option test: " + option);
+					});
+				});
+			});
+		});
+	});
+});*/
+
+
+// Test external URL's and check that input fields are found and the password generator icon is visible
+// This site could also be internal and included in the test folder.
+test.describe('Testing external sites', function() {
+	test.it('Test input fields', function() {
+		wd.get("https://www.github.com/login");
```

Comment:
It would be interesting to use the [test form at keepass.info](https://keepass.info/help/kb/testform.html) instead since it is less likely to change than the github login page.

Reply:
Good idea! Also, of course it's possible to create a localhost based test pages or use the ones from w3school.com.

Reply:
Still, we need to have test for pages that create the login form from a div that is hidden or created dynamically.

Reply:
We can host our own too. I personally prefer to bundle any test pages in with the test suite. Don't rely on "live" sites for your testing.